norddrop/src/norddrop.udl

```diff
@@ -91,128 +174,325 @@ interface Event {
     Incoming(sequence<IncomingPath> paths);
+
+	/// The transfer is incoming, meaning we are the one who sends the files
     Outgoing(sequence<OutgoingPath> paths);
 };
 
+/// Transfer and files in it contain history of states that can be used to
+/// replay what happened and the last state denotes the current state of the
+/// transfer.
 dictionary TransferInfo {
+	/// Transfer UUID
     string id;
+
+	/// The creation time as a UNIX timestamp in milliseconds.
     i64 created_at;
+
+	/// Peer's IP address
     string peer;
+
+	/// History of transfer states
     sequence<TransferState> states;
+
+	/// The transfer type description
     TransferKind kind;
 };
 
 interface NordDrop {
+	/// Create a new instance of norddrop. This is a required step to work
+	/// with API further
+	///
+	/// # Arguments
+	/// * `event_cb` - Event callback
+	/// * `logger` - Logger callback
+	/// * `key_store` - Fetches peer's public key and provides own private key. 
 	[Throws=Error]
 	constructor(EventCallback event_cb, KeyStore key_store, Logger logger);
 
+	/// Starts libdrop
+	///
+	/// # Arguments
+	/// * `addr` - Address to listen on
+	/// * `config` - configuration
+	///
+	/// # Configuration Parameters
+	///
+	/// * `dir_depth_limit` - if the tree contains more levels then the error is
+	/// returned.
+	///
+	/// * `transfer_file_limit` - when aggregating files from the path, if this
+	/// limit is reached, an error is returned.
+	///
+	/// * `moose_event_path` - moose database path.
+	///
+	/// * `moose_prod` - moose production flag.
+	///
+	/// * `storage_path` - storage path for persistence engine.
+	///
+	/// * `checksum_events_size_threshold_bytes` - emit checksum events only if file
+	///   is equal or greater than this size. If omited, no checksumming events are
+	///   emited.
+	///
+	/// # Safety
+	/// The pointers provided must be valid
 	[Throws=Error]
 	void start([ByRef] string addr, Config config);
 
+	/// Stop norddrop instance
 	[Throws=Error]
 	void stop();
 
+	/// Purge transfers from the database
+	///
+	/// # Arguments
+	/// * `transfer_ids` - array of transfer UUIDs
 	[Throws=Error]
 	void purge_transfers([ByRef] sequence<string> transfer_ids);
 
+	/// Purge transfers from the database until the given timestamp
+	///
+	/// # Arguments
+	/// * `until` - Unix timestamp in milliseconds
 	[Throws=Error]
 	void purge_transfers_until(i64 until);
 
+	/// Get transfers from the database
+	///
+	/// # Arguments
+	/// * `since_timestamp` - UNIX timestamp in milliseconds
 	[Throws=Error]
 	sequence<TransferInfo> transfers_since(i64 since);
 
+	/// Initialize a new transfer with the provided peer and descriptors
+	///
+	/// # Arguments
+	/// * `peer` - Peer address.
+	/// * `descriptors` - transfer file descriptors.
+	///
+	/// # Returns
+	/// A String containing the transfer UUID.
 	[Throws=Error]
 	string new_transfer([ByRef] string peer, [ByRef] sequence<TransferDescriptor> descriptors);
 
+	/// # Finishes the transfer from either side
```

Comment:
I sort of just noticed but I think `finalize` would be even better as there's no way to misinterpret it, what do you think?

Reply:
Yeah, it makes sense :+1:

Reply:
Done